README.md

```diff
@@ -1,121 +1,338 @@
+
+#### Examples
+
+##### CommonJS
+
+```javascript
+StorageManager = require('brainstem-js').StorageManager;
+Users = require('./collections/users');
+Posts = require('./collections/posts');
+);
+
+storageManger = StorageManager.get();
+storageManager.addCollection('users', Users);
+storageManager.addCollection('posts', Posts);
+storageManager.addCollection('comments', Comments);
+```
+
+##### Vanilla JavaScript
+
+```javascript
+Application = {};
+
+Application.storageManager = Brainstem.StorageManager.get();
+Application.storageManager.addCollection('users', Application.Collections.Users);
+Application.storageManager.addCollection('posts', Application.Collections.Posts);
+Application.storageManager.addCollection('comments', Application.Collections.Comments);
+```
+
+<br>
+
+
+#### *Note: all preceding examples assume a CommonJS environment, however the same functionality applies to vanilla JavaScript environments*
+
+---
+
+
+
+### Fetching data
+
+Brainstem.js extends the Backbone `fetch` API so requesting data from a Brainstem API should be familiar to fetching data from any RESTful API using just Backbone.
+
+#### Models
+
+In addition to basic REST requests, the Brainstem.js model `fetch` method supports an `include` option to side-load associated model data.
+
+##### Example
+
+```javascript
+Post = require('./models/post');
+
+new Post({ id: 1 }).fetch({ include: ['user', 'comments'] })
+  .done(/* handle result */)
+  .fail(/* handle error */);
+```
+
+#### Collections
+
+In addition to basic REST requests, the Brainstem.js model `fetch` method supports additional Brainstem options:
+
+- Pagination using `page` and `perPage`, or `offset` and `limit`
+- Filtering using `filters` object
+- Ordering user `order` string
+
+##### Example
+
+```javascript
+Posts = require('./collections/posts');
+
+new Posts().fetch({
+  page: 1,
+  perPage: 10,
+  order: 'date:desc',
+  filters: {
+    title: 'collections',
+    description: 'fetching'
+ })
+}).done(/* handle result */)
+  .fail(/* handle error */);
+```
+
+### Accessing Model Associations
+
+##### Example
+
+```javascript
+Post = require('./models/post');
+
+var user;
+var comments;
+
+new Post({ id: 1 }).fetch({ include: ['user', 'comments'] })
+  .done(function (post) {
+  	user = post.get('user');
+  	comments = post.get('comments');
+  });
+  
+console.log('user');
+// User [BackboneModel]
+
+console.log('comments');
+// Comments [BackboneCollection]
+```
+
+### Manipulating Collections
+
+#### Filter Scoping
+
+Brainstem.js collections provide a filter scoping mechanism that allows a base scope to be defined either by providing base `filter` and `order` options to the Brainste.js Collection constructor, or by passing said options to the *first* `fetch` call.
```

Comment:
I don't think is is up-to-date: `firstFetchOptions` are set only in `initialize`?